src/dmd/dsymbolsem.d

```diff
@@ -2107,6 +2108,64 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         attribSemantic(cd);
     }
 
+    override void visit(CPPNamespaceDeclaration ns)
+    {
+        Identifier identFromSE (StringExp se)
+        {
+            const sident = se.toStringz();
+            if (!sident.length || !Identifier.isValidIdentifier(sident))
+            {
+                ns.exp.error("expected valid identifer for C++ namespace but got `%.*s`",
```

Comment:
Wasn't the motive of this syntax that D identifiers don't exactly match C++? [15512](https://issues.dlang.org/show_bug.cgi?id=15512).

Reply:
Yes, it's very important.

Reply:
Then checking the validity here defeats the point.

Reply:
This code was adapted from what's existing. I also had a "Wait, what?" moment until I checked the [code for `isValidIdentifier`](https://github.com/dlang/dmd/blob/140937df3da88cce0a3cdeb7a5056f0a0e0aad96/src/dmd/identifier.d#L268-L288). It does not check if the identifier is one of the reserved ones.